pkgs/build-support/agda/lib.nix

```diff
@@ -6,7 +6,7 @@
   * interfaceFile "Everything.agda" == "Everything.agdai"
   * interfaceFile "src/Everything.lagda.tex" == "src/Everything.agdai"
   */
-  interfaceFile = agdaFile: lib.head (builtins.match ''(.*\.)l?agda(\.(md|org|rst|tex|typ))?'' agdaFile) + "agdai";
+  interfaceFile = agdaVersion: agdaFile: "./_build/" + agdaVersion + "/agda/" + lib.head (builtins.match ''./(.*\.)l?agda(\.(md|org|rst|tex|typ))?'' agdaFile) + "agdai";
```

Comment:
```suggestion
  interfaceFile = agdaVersion: agdaFile: "_build/" + agdaVersion + "/agda/" + lib.head (builtins.match ''(.*\.)l?agda(\.(md|org|rst|tex|typ))?'' agdaFile) + "agdai";
```

Reply:
In `pkgs/development/libraries/agda/1lab/default.nix:25` the assumption of a leading `./` is broken. The leading `./` could be made optional to be more robust I guess. However, I think that means that the everything file (`src/index.lagda.md`) isn't exlcuded as well. I guess in this case this is actually desired.

I think I will add another commit to make the `everythingFile` parameter more robust and add a parameter to disable removal of this file.

Reply:
Ugh, I always forget that `find -path` compares strings and not paths 🙄.

What's the point of excluding the everything file anyway? Can we just leave it in?

Reply:
Excerpt from `git log`:

> commit 226299e1a2e8dbc9ee8b10042182d8a1f47d7f16
> Author: Alexander Ben Nasrallah <me@abn.sh>
> Date:   Mon Jan 18 20:01:31 2021 +0100
> 
>     agdaPackages.mkDerivation: don't install Everything module
>     
>     The Everthing module is not part of a library and should therefore
>     not be copied to the nix store.
>     
>     This is particularly bad, if the Everything module is defined in
>     an agda library included directory, e.g. consider an agda-lib with
>     
>         include: .
>     
>     and Everything.agda in the project root (.), in which case the
>     Everything module would become part of the library.
>     If multiple such projects are in the dependency tree, the Everything
>     module becomes ambiguous and the build would fail.

However, in the case of 1lab and agdarsec the everything file kind of act as documentation so I'm not sure what should be done there.

Reply:
As a fix I exchanged the `-path` exclusion from `find` by removing the file directly before installing. This should make it more robust in parsing the file path and prevent the builder from continuing in case the exclusion fails.

Reply:
Thanks, this seems like a sensible fix.

In the case of 1lab, upstream recently abandoned the "index as Everything file" idea, so future updates will have to find another way to generate an Everything file (and in that case we will want it to be removed), so all is good.

Reply:
Well, except now `cubical` fails to build because the default `Everything.agda` isn't there. We should not fail if `rm` fails; perhaps add `-f`?

Reply:
Oh, right. I missed that.
However, I think it's still a good idea to catch potential errors (e.g., naming changes regarding the interface file). Hence, I decided to move the `rm` into `buildPhase`. This ensures that there will always be an everything file, otherwise the build would fail anyways.

Do you think that's a reasonable compromise?

Reply:
Yes that's perfect.